doc/main/tbb_userguide/How_Does_Task_Scheduler_Works.rst

```diff
@@ -0,0 +1,43 @@
+.. _How_Does_Task_Scheduler_Works.rst:
+
+How Does Task Scheduler Works
+=============================
+
+The scheduler runs tasks in a way that tends to minimize both memory 
+demands and cross-thread communication. The intuition is that a balance 
+must be reached between depth-first and breadth-first execution. 
+Assuming that the tree is finite, depth-first is best for sequential 
+execution for the following reasons:
```

Comment:
It seems tree decomposition oriented. While TBB scheduler and tasking approach was (almost) tree structure oriented, the oneTBB scheduler is completely unaware of task dependencies and we need to think how to explain it better. Maybe we can first say about organization of queues and so on but then explain how it is mapped on tree decomposition that is widely used in parallel algorithms but not in flow graph.

Reply:
changed